commands/operator-sdk/cmd/build.go

```diff
@@ -42,12 +54,72 @@ For example:
 `,
 		Run: buildFunc,
 	}
+	buildCmd.Flags().BoolVarP(&enableTests, "enable-tests", "e", false, "Enable in-cluster testing by adding test binary to the image")
+	buildCmd.Flags().StringVarP(&testLocationBuild, "test-location", "t", "./test/e2e", "Location of tests")
+	buildCmd.Flags().StringVarP(&namespacedManBuild, "namespaced", "n", "deploy/operator.yaml", "Path of namespaced resources for tests")
+	return buildCmd
+}
+
+func verifyDeploymentImage(yamlFile []byte, imageName string) string {
+	warningMessages := ""
+	yamlSplit := bytes.Split(yamlFile, []byte("\n---\n"))
+	for _, yamlSpec := range yamlSplit {
+		yamlMap := make(map[string]interface{})
+		err := yaml.Unmarshal(yamlSpec, &yamlMap)
+		if err != nil {
+			fmt.Printf("WARNING: Could not unmarshal yaml namespaced spec")
+			return ""
+		}
+		kind, ok := yamlMap["kind"].(string)
+		if !ok {
+			log.Fatal("Yaml manifest file contains a 'kind' field that is not a string")
+		}
+		if kind == "Deployment" {
```

Comment:
The `verifyDeploymentImage` function checks the entire namespaced manifest file, and while the most basic case would be just a single deployment for the operator, it could have multiple deployments or other objects, such as configmaps. We use `kind=='Deployment'` to ignore other resources and just check deployments.